### PR TITLE
[Breaking] Fix gitlab_project and gitlab_project_share relationship

### LINKED ISSUE #219: gitlab_project_share_group not working
Using the `gitlab_project_share_group` resource is not retained correctly in Terraform state as at any subsequent `plan` or `apply` the resource appears as new, while in turn it seems it is converted into the `gitlab_project.shared_with_groups` block, causing a false configuration drift.

### Terraform Version
```
$ terraform -v
Terraform v0.12.9
+ provider.gitlab v2.3.0
```

### Affected Resource(s)
Please list the resources as a list, for example:
- gitlab_project_share_group
- gitlab_project

### Terraform Configuration Files
```hcl
resource "gitlab_project_share_group" "project_share_group" {
  for_each                  = var.share_with_groups
  project_id                = gitlab_project.project[0].id
  group_id                  = each.key
  access_level              = each.value
}
```

### Expected Behavior
The resource should apply and retained in Terraform state, without being detected as a new resource at the next `plan` or `apply`.

### Actual Behavior
The resource is created with no errors but the next `plan` or `apply` don't see this resource in the Terraform state, instead it appears as a configuration drift in the `gitlab_project` resource, which appears to have a new `shared_with_groups` new block.

### Steps to Reproduce
1. create a `gitlab_project` resource and one `gitlab_project_share_group`
2. `terraform apply`
3. `terraform plan`

### PR DESCRIPTION
**Pull request by armsnyder**
- resource/gitlab_project: remove `shared_with_groups` attribute
- resource/gitlab_project_share: fix update failing to change `access_level`

Part of #411 preparation

See discussion in #416 

Fixes #219

### CODE REVIEW

File: gitlab/data_source_gitlab_projects.go [hunk 1]
```diff
@@ -850,3 +850,19 @@ func dataSourceGitlabProjectsRead(d *schema.ResourceData, meta interface{}) (err
 	}
 	return err
 }
+
+func flattenSharedWithGroupsOptions(project *gitlab.Project) []interface{} {
```
Comment (armsnyder) [inline gitlab/data_source_gitlab_projects.go:854]: This function is moved from its original home in `gitlab/resource_gitlab_project.go`.

File: gitlab/resource_gitlab_project_share_group.go [hunk 2]
```diff
@@ -104,26 +103,6 @@ func projectIdAndGroupIdFromId(id string) (string, int, error) {
 	return projectId, groupId, nil
 }
 
-func resourceGitlabProjectShareGroupUpdate(d *schema.ResourceData, meta interface{}) error {
```
Comment (armsnyder) [inline gitlab/resource_gitlab_project_share_group.go:107]: After making `access_level` `ForceNew`, the updater is no longer needed.

File: gitlab/resource_gitlab_project_share_group_test.go [hunk 3]
```diff
@@ -12,138 +12,109 @@ import (
...
-  name = "foo%d"
-  description = "Terraform acceptance tests"
-  visibility_level ="public"
+resource "gitlab_group" "test" {
+  name = "%[1]s"
+  path = "%[1]s"
+}
 
-  lifecycle {
-    ignore_changes = [
```
Comment (armsnyder) [inline gitlab/resource_gitlab_project_share_group_test.go:136]: This `ignore_changes` was hiding the bug 😦

**Review by roidelapluie — APPROVED** [review]
LGTM, thanks!

**Pull request merged**